staging/src/k8s.io/kubectl/pkg/cmd/config/set.go

```diff
@@ -38,13 +42,17 @@ type setOptions struct {
 	propertyName  string
 	propertyValue string
 	setRawBytes   cliflag.Tristate
+	jsonPath      bool
+	deduplicate   bool
```

Comment:
This `deduplicate` feature seems like it might be out of scope for this PR?

Reply:
Yeah this may not be necessary really. I originally wrote the functions that support this flag to allow me to extend the existing syntax, which didn't support the idea of being able to set a specific value in an array or filtering results, but this implementation using jsonpath now supports that, so this may be pointless I guess. If others could chime in if they think this is necessary or not I'd appreciate it.

It's also a bit of a bad name for the flag because the it uses the `k8s.io/apimachinery/pkg/util/sets` which also sorts in addition to deduplicating.